### PR TITLE
Generate sitemap

### PR DESCRIPTION
**Pull request by lemieux**
This PR adds sitemap generation for better result in search engines.

@alavers @mspensieri @dannytranlx

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -1 +1,40 @@
+// TODO : Figure out why there's a possible EventEmitter memory leak instead of blindly ignore it.
```
Comment (dannytranlx) [inline gatsby-node.js]: Not again 😩

File: package.json [hunk 2]
```diff
@@ -23,6 +24,7 @@
     "react-headroom": "^2.1.2",
     "react-helmet": "^3.2.2",
     "react-router-bootstrap": "^0.23.1",
+    "sitemap": "^1.8.2",
```
Comment (dannytranlx) [inline package.json]: nit: Not really related, do we need strict versions in package.json?

Comment (lemieux) [inline package.json]: yeah I'll fix them

Comment (lemieux) [inline package.json]: done

File: gatsby-node.js [hunk 3]
```diff
@@ -1 +1,40 @@
...
+}
+
+function generateSiteMap(pages) {
+    const sitemap = sm.createSitemap({
+        hostname: 'https://docs.smooch.io',
+        cacheTime: '60000',
+        urls: pagesToSitemap(pages),
+    });
+
+    console.log('Generating sitemap.xml');
```
Comment (alavers) [inline gatsby-node.js]: nit, log the actual path of the sitemap.xml

File: gatsby-node.js [hunk 4]
```diff
@@ -1 +1,40 @@
...
+import sm from 'sitemap' ;
+
+// Sitemap generation taken from suggestions in https://github.com/gatsbyjs/gatsby/issues/12
+
+function pagesToSitemap(pages) {
+    const urls = pages.map(({path}) => {
+        if (path !== undefined) {
+            return {
+                url: path,
+                changefreq: 'daily',
```
Comment (alavers) [inline gatsby-node.js]: how do changefreq and priority affect this?

Comment (lemieux) [inline gatsby-node.js]: `changefreq` is to tell crawlers how frequently they should look for changes and `priority` is to tell the crawlers which pages are more important.. so it doesn't really matter actually in our case.

Comment (lemieux) [inline gatsby-node.js]: http://www.sitemaps.org/protocol.html

**Pull request merged**